src/library/scala/collection/immutable/List.scala

```diff
@@ -477,6 +482,7 @@ sealed abstract class List[+A]
 @SerialVersionUID(4L)
 case class :: [+A](override val head: A, private[scala] var next: List[A @uncheckedVariance]) // sound because `next` is used only locally
```

Comment:
Is this comment really true given the reachability of the var in the generated bytecode.

Reply:
You're right, this doesn't state the assumption that people won't mutate this from Java with `tl_$eq`. But that's a general problem we have (e.g. trait vals are mutable from Java, too)